lms/djangoapps/courseware/user_state_client.py

```diff
@@ -192,12 +134,23 @@ def get_many(self, username, block_keys, scope=Scope.user_state, fields=None):
         modules = self._get_student_modules(username, block_keys)
         for module, usage_key in modules:
             if module.state is None:
-                state = {}
-            else:
-                state = json.loads(module.state)
-            yield (usage_key, state)
+                continue
+
+            state = json.loads(module.state)
+
+            # If the state is the empty dict, then it has been deleted, and so
+            # conformant UserStateClients should treat it as if it doesn't exist.
+            if state == {}:
+                continue
+
+            if fields is not None:
+                state = {
+                    field: state[field]
+                    for field in fields
+                    if field in state
+                }
```

Comment:
I realize I just need to get over it - but this assignment to `state` while using the different `state` in the dict comprehension raises alarm bells from past languages. Again - my own problem...